src/main.rs

```diff
@@ -53,24 +54,20 @@ fn read_input<T: AsRef<str>>(filename: T) -> std::io::Result<(PathBuf, String)>
 }
 
 fn process_arguments(size: TerminalSize, args: Arguments) -> Result<(), Box<dyn Error>> {
-    if args.detect_only {
-        println!("Terminal: {}", args.terminal_capabilities.name);
-        Ok(())
-    } else {
```

Comment:
Please don't move this part out of `process_arguments`.  

`process_arguments` should really, well, process all arguments :) so please find a way to keep `args.detect_only` inside this method.

Reply:
done